torch/distributed/distributed_c10d.py

```diff
@@ -1875,7 +1961,7 @@ def _check_for_nccl_backend(group):
 
     return (
         is_nccl_available() and
-        isinstance(pg, ProcessGroupNCCL)
+        pg.name() == Backend.NCCL
```

Comment:
Are we treating `pg` as a `Backend` here?
Should it be a `ProcessGroup`? Then you can indeed use the pg_to_backend_config map here.

Reply:
pg is a `ProcessGroup`, good point, ill just use `get_backend_config()`